libecho/src/echo.cpp

```diff
@@ -30,7 +32,8 @@ std::vector<uint8_t> echo::receive() {
     const size_t atOnce = info.periodSize;
     char *array = new char[atOnce];
```

Comment:
unique_ptr zamiast gołych wskaźników?

https://stackoverflow.com/questions/21377360/proper-way-to-create-unique-ptr-that-holds-an-allocated-array

Reply:
Tak jak wyżej - mogę zmienić ale to trochę bezcelowe

Reply:
mam na myśli przede wszystkim ogólne stosowanie unique_ptr w kodzie zamiast gołych wskaźników; gdy stosujemy to w jednym miejscu, to powinno być ok, ale przy większej ilości możemy się pogubić; dodatkowo sprytne wskaźniki chronią nas przed gubieniem pamięci w przypadku jakichś dziwnych bugów w Qt (np. skutkujących wyjątkami)

Reply:
A to ja się w pełni z tym zgadzam i bardzo popieram używanie smart pointerów. W tym jednym miejsciu nie widzę sensu zmian - to jedynie tymczasowy kod, zaraz go nie będzie. Przejrzałeminne miejsca w kodzie gdzie trzymam surowe wskaźniki i poza jednym wyjątkiem (QCoreApplication w klasie QTInitializer) niestety nie widzę jak mógłbym je zmienić na smart pointery (StreamType w AudioStream musi być surowym wskaźnikiem, żeby dało sie go przekazać do connect(), QIODevice * jest zwracane przez metodę start() QAudioInput/Output i nie my nim zarządzamy).